jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java

```diff
@@ -15,35 +13,31 @@
  */
 public final class GenericJBossMarshaller extends AbstractJBossMarshaller {
 
+   private final ClassLoader classLoader;
+
    public GenericJBossMarshaller() {
-      super();
-      baseCfg.setClassResolver(
-            new DefaultContextClassResolver(this.getClass().getClassLoader()));
+      this(null, null);
    }
 
    public GenericJBossMarshaller(ClassLoader classLoader) {
-      super();
-      baseCfg.setClassResolver(
-            new DefaultContextClassResolver(classLoader != null ? classLoader : this.getClass().getClassLoader()));
-   }
-
-   /**
-    * @deprecated Use {@link #GenericJBossMarshaller(ClassWhiteList)} instead
-    */
-   @Deprecated
-   public GenericJBossMarshaller(List<String> whitelist) {
-      this(new ClassWhiteList(whitelist));
+      this(classLoader, null);
    }
 
    public GenericJBossMarshaller(ClassWhiteList classWhiteList) {
-      super();
-      baseCfg.setClassResolver(
-            new CheckedClassResolver(classWhiteList, this.getClass().getClassLoader()));
+      this(null, classWhiteList);
    }
 
    public GenericJBossMarshaller(ClassLoader classLoader, ClassWhiteList classWhiteList) {
       super();
-      baseCfg.setClassResolver(new CheckedClassResolver(classWhiteList, classLoader));
+      this.classLoader = classLoader == null ? getClass().getClassLoader() : classLoader;
```

Comment:
Not doing ```getClass().getClassLoader()``` and just leaving it null would default to TCCL. Not sure which one is best.

Reply:
I think we should stick with the current behaviour of `getClass().getClassLoader()` to avoid any unexpected surprises.

Reply:
ok.